InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -285,6 +297,42 @@ TupleSet AstQuery::usesQuery(TupleSet input)
 	return result;
 }
 
+TupleSet AstQuery::typeFilter(TupleSet input)
+{
+	using SymbolType = Model::Node::SymbolType;
+
+	SymbolType expectedSymbolType = SymbolType::VARIABLE;
+	// FIXME don't use nodetype argument as it is the actual type:
+	QString expectedType;
+	QStringList arguments;
+	auto typeArgument = argument(NODETYPE_ARGUMENT_NAMES[1]);
+	// NOTE: no spaces allowed for this argument.
```

Comment:
This is a limitation of the QCommandLineParser, which is used to parse all arguments, since it splits by space I guess.

As for the examples:
Filter out fields:
Only keep int fields: ```type -t=int```
Only keep fields of type Node pointer: ```type -t=Node*```
 
Filter out methods
Filter methods with any return type and no arguments: ```type -t=_()```
Filter methods with int return type and 2 arguments of any type: ```type -t=int(_,_)```